dashboard/app/jobs.go

```diff
@@ -125,32 +147,36 @@ func addTestJob(c context.Context, bug *Bug, bugKey *db.Key, bugReporting *BugRe
 		// We can get 2 emails for the same request: one direct and one from a mailing list.
 		// Filter out such duplicates (for dup we only need link update).
 		var jobs []*Job
-		keys, err := db.NewQuery("Job").
-			Ancestor(bugKey).
-			Filter("ExtID=", extID).
-			GetAll(c, &jobs)
-		if len(jobs) > 1 || err != nil {
-			return fmt.Errorf("failed to query jobs: jobs=%v err=%v", len(jobs), err)
+		var keys []*db.Key
+		var err error
+		if args.extID != "" {
+			keys, err = db.NewQuery("Job").
+				Ancestor(args.bugKey).
+				Filter("ExtID=", args.extID).
+				GetAll(c, &jobs)
+			if len(jobs) > 1 || err != nil {
+				return fmt.Errorf("failed to query jobs: jobs=%v err=%v", len(jobs), err)
+			}
 		}
 		if len(jobs) != 0 {
 			// The job is already present, update link.
 			deletePatch = true
 			existingJob, jobKey := jobs[0], keys[0]
-			if existingJob.Link != "" || link == "" {
+			if existingJob.Link != "" || args.link == "" {
 				return nil
 			}
-			existingJob.Link = link
+			existingJob.Link = args.link
 			if _, err := db.Put(c, jobKey, existingJob); err != nil {
 				return fmt.Errorf("failed to put job: %v", err)
 			}
 			return nil
 		}
 		// Create a new job.
-		jobKey := db.NewIncompleteKey(c, "Job", bugKey)
+		jobKey := db.NewIncompleteKey(c, "Job", args.bugKey)
 		if _, err := db.Put(c, jobKey, job); err != nil {
 			return fmt.Errorf("failed to put job: %v", err)
 		}
-		return markCrashReported(c, job.CrashID, bugKey, now)
+		return markCrashReported(c, job.CrashID, args.bugKey, now)
```

Comment:
It may be reasonable if markCrashReported will update crash.LastStateCheck based on an additional flag.
Maybe it also makes sense to save "isCrashRetest" flag in the Job object, since we will need to calculate it here and it will be needed when we handle job completion.

Reply:
>  it will be needed when we handle job completion.

Can you please elaborate?

Reply:
Add `job.IsReproRetest bool`, fill it here.
Pass `job.IsReproRetest` to `markCrashReported`, which will update `crash.LastStateCheck = now` for such jobs.
Use `job.IsReproRetest` instead of `isRetestReproJob` function.

Reply:
Tbh I'm not sure if this change is actually needed in this case. Currently we update `LastReproRetest` when we have actually retested it and it works perfectly fine with the current job-deduplication scheme (only one per manager at a time).